Gradle-ify the project, update used APIs make it work with the current CurseForge system

iTitus opened the pull request:
I did this in my fork for myself, but I figured you could use the general idea.

The new link is:
`https://addons-ecs.forgesvc.net/api/v2/addon/<project id>/file/<file id>`
If you want modpack data just set the project id to 0.

This returns JSON data, to get the real download url you can either parse the JSON or access `/download-url` on that resource.

The returned URL can be used to `GET` the required file.

iTitus committed:
Gradle-ify the project

iTitus committed:
Use NIO

iTitus committed:
Finally make it work

iTitus committed:
Remove test deps

iTitus committed:
Downgrade to Java 8

iTitus committed:
update gradle

stefnotch commented:
@iTitus Could you build it and publish the jar in your fork?

iTitus committed:
update gradle

iTitus committed:
enable fat jars

iTitus committed:
fix resourcepacks loading

iTitus commented:
@stefnotch here you go
https://github.com/iTitus/CMPDL/releases/tag/v1.3.0

stefnotch commented:
Woah, that's the fastest positive response I've ever gotten to such a request! Thank you very much!

Chocohead committed:
Fix crashing from cleaning the extract dir (#1)

If the extraction directory doesn't exist then Files#walk will crash trying to get the attributes of a non-existent path

iTitus committed:
Update Gradle and increment version

iTitus committed:
multi-threaded downloads and copying

iTitus committed:
update shadow plugin

iTitus committed:
update

iTitus committed:
fix shadow

dependabot[bot] committed:
Bump moshi from 1.11.0 to 1.12.0

Bumps [moshi](https://github.com/square/moshi) from 1.11.0 to 1.12.0.
- [Release notes](https://github.com/square/moshi/releases)
- [Changelog](https://github.com/square/moshi/blob/master/CHANGELOG.md)
- [Commits](https://github.com/square/moshi/compare/moshi-parent-1.11.0...parent-1.12.0)

Signed-off-by: dependabot[bot] <support@github.com>

iTitus committed:
Merge pull request #3 from iTitus/dependabot/gradle/com.squareup.moshi-moshi-1.12.0

Bump moshi from 1.11.0 to 1.12.0

iTitus committed:
update setup-java action

iTitus committed:
Update gradle, java & shadow plugin

iTitus committed:
update wrapper

iTitus committed:
update gradle

iTitus committed:
update gradle

dependabot[bot] committed:
Bump com.github.johnrengelman.shadow from 7.0.0 to 7.1.0

Bumps com.github.johnrengelman.shadow from 7.0.0 to 7.1.0.

---
updated-dependencies:
- dependency-name: com.github.johnrengelman.shadow
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

iTitus committed:
Merge pull request #9 from iTitus/dependabot/gradle/com.github.johnrengelman.shadow-7.1.0

Bump com.github.johnrengelman.shadow from 7.0.0 to 7.1.0

dependabot[bot] committed:
Bump moshi from 1.12.0 to 1.13.0

Bumps [moshi](https://github.com/square/moshi) from 1.12.0 to 1.13.0.
- [Release notes](https://github.com/square/moshi/releases)
- [Changelog](https://github.com/square/moshi/blob/master/CHANGELOG.md)
- [Commits](https://github.com/square/moshi/compare/moshi-parent-1.12.0...moshi-parent-1.13.0)

---
updated-dependencies:
- dependency-name: com.squareup.moshi:moshi
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

iTitus committed:
Merge pull request #10 from iTitus/dependabot/gradle/com.squareup.moshi-moshi-1.13.0

Bump moshi from 1.12.0 to 1.13.0

dependabot[bot] committed:
Bump com.github.johnrengelman.shadow from 7.1.0 to 7.1.1

Bumps com.github.johnrengelman.shadow from 7.1.0 to 7.1.1.

---
updated-dependencies:
- dependency-name: com.github.johnrengelman.shadow
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

iTitus committed:
Merge pull request #11 from iTitus/dependabot/gradle/com.github.johnrengelman.shadow-7.1.1

Bump com.github.johnrengelman.shadow from 7.1.0 to 7.1.1

dependabot[bot] committed:
Bump com.github.johnrengelman.shadow from 7.1.1 to 7.1.2

Bumps com.github.johnrengelman.shadow from 7.1.1 to 7.1.2.

---
updated-dependencies:
- dependency-name: com.github.johnrengelman.shadow
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

iTitus committed:
Merge pull request #12 from iTitus/dependabot/gradle/com.github.johnrengelman.shadow-7.1.2

iTitus committed:
update deps

dependabot[bot] committed:
Bump actions/checkout from 2 to 3

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

iTitus committed:
Merge pull request #13 from iTitus/dependabot/github_actions/actions/checkout-3

Bump actions/checkout from 2 to 3

dependabot[bot] committed:
Bump actions/cache from 2 to 3

Bumps [actions/cache](https://github.com/actions/cache) from 2 to 3.
- [Release notes](https://github.com/actions/cache/releases)
- [Commits](https://github.com/actions/cache/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

iTitus committed:
Merge pull request #14 from iTitus/dependabot/github_actions/actions/cache-3

Bump actions/cache from 2 to 3

dependabot[bot] committed:
Bump actions/upload-artifact from 2 to 3

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 2 to 3.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/setup-java from 2 to 3

Bumps [actions/setup-java](https://github.com/actions/setup-java) from 2 to 3.
- [Release notes](https://github.com/actions/setup-java/releases)
- [Commits](https://github.com/actions/setup-java/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-java
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

iTitus committed:
Merge pull request #16 from iTitus/dependabot/github_actions/actions/setup-java-3

Bump actions/setup-java from 2 to 3

iTitus committed:
Merge pull request #15 from iTitus/dependabot/github_actions/actions/upload-artifact-3

Bump actions/upload-artifact from 2 to 3

dependabot[bot] committed:
Bump github/codeql-action from 1 to 2

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 1 to 2.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v1...v2)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

iTitus committed:
Merge pull request #17 from iTitus/dependabot/github_actions/github/codeql-action-2

Bump github/codeql-action from 1 to 2

5yn74x committed:
Using the new CurseForge API

Signed-off-by: 5yn74x <30578607+5yn74x@users.noreply.github.com>

5yn74x committed:
download version using curseforge.com/minecraft/modpacks/<slug>/files/<ID> url

Signed-off-by: 5yn74x <30578607+5yn74x@users.noreply.github.com>

dependabot[bot] committed:
Bump moshi from 1.13.0 to 1.14.0

Bumps [moshi](https://github.com/square/moshi) from 1.13.0 to 1.14.0.
- [Release notes](https://github.com/square/moshi/releases)
- [Changelog](https://github.com/square/moshi/blob/master/CHANGELOG.md)
- [Commits](https://github.com/square/moshi/compare/moshi-parent-1.13.0...1.14.0)

---
updated-dependencies:
- dependency-name: com.squareup.moshi:moshi
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

iTitus committed:
Merge pull request #19 from iTitus/dependabot/gradle/com.squareup.moshi-moshi-1.14.0

Bump moshi from 1.13.0 to 1.14.0

5yn74x committed:
Use java version <= 1.8

5yn74x committed:
Update src/main/java/vazkii/cmpdl/CMPDL.java

Co-authored-by: iTitus <iTitus@users.noreply.github.com>

iTitus committed:
Merge pull request #18 from 5yn74x/fix

New CurseForge API
